huaweicloud/services/apigateway/resource_huaweicloud_api_gateway_environment.go

```diff
@@ -0,0 +1,149 @@
+package apigateway
+
+import (
+	"context"
+	"log"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/chnsz/golangsdk"
+	"github.com/chnsz/golangsdk/openstack/apigw/shared/v1/environments"
+
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/common"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils"
+)
+
+func ResourceAPIGatewayEnvironment() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceAPIGatewayEnvironmentCreate,
+		ReadContext:   resourceAPIGatewayEnvironmentRead,
+		UpdateContext: resourceAPIGatewayEnvironmentUpdate,
+		DeleteContext: resourceAPIGatewayEnvironmentDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"description": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"create_time": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceAPIGatewayEnvironmentCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+	apigClient, err := cfg.ApiGatewayV1Client(region)
+	if err != nil {
+		return diag.Errorf("error creating API gateway client: %s", err)
```

Comment:
```suggestion
		return diag.Errorf("error creating shared API gateway client (v1) : %s", err)
```
Others also recommend printing this way.